helperAPI.py

```diff
@@ -352,17 +355,28 @@ def is_up_to_date(remote, branch):
     try:
         g = git.cmd.Git()
         ls_remote = g.ls_remote(remote, branch)
-        remote_hash = str(ls_remote.split("\t")[0])
+        remote_hash = ls_remote.split("\n")
+        wanted_remote = f"refs/heads/{branch}"
+        for line in remote_hash:
+            if wanted_remote in line:
+                remote_hash = line.split("\t")[0]
+                break
+        if isinstance(remote_hash, list):
+            remote_hash = ""
+            is_fork = True
+            raise Exception(
+                f"Branch {branch} not found in remote {remote}. Perhaps you are on a fork?"
+            )
         if local_commit == remote_hash:
             up_to_date = True
             print(f"You are up to date with {remote}/{branch}")
     except Exception as e:
         print(f"Error running ls-remote: {e}")
-    if not up_to_date:
+    if not up_to_date and not is_fork:
         if remote_hash == "":
             remote_hash = "NOT FOUND"
         print(
-            f"WARNING: YOU ARE OUT OF DATE. Please run 'git pull {remote} {branch}' to update. Local hash: {local_commit}, Remote hash: {remote_hash}"
+            f"WARNING: YOU ARE OUT OF DATE. Please run 'git pull' to update from {remote}/{branch}. Local hash: {local_commit}, Remote hash: {remote_hash}"
```

Comment:
revert all of the changes here. we will run into issues since we're running a fork

Reply:
resolved